### PR TITLE
Add Pull Request Checklist

### LINKED ISSUE #1458: Add checklist to Pull Request template
#### Summary

I think we should add a checklist to our pull request template.

#### Why do we need this?

To formalize the process a bit, make sure that reviewers look at the things that need to be looked at and to make sure we're not skipping any steps, even when we're in a hurry.

#### What is missing? What do you want to see?

- We used to have a **Release Notes** section in the Pull Request template, that is now replaced with explicit changes in `CHANGELOG.md`, but it's easy to forget to do this, and it's easy for a reviewer not to check this.
- We need to check for commit messages in case someone forgot to install the hooks (actually CI should do this).
- We want the relevant documentation to be updated when things change.
- ...

#### Can you do this yourself and submit a Pull Request?

Sure, let's discuss what exactly should be on the checklist, and then update the template.

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This Pull Request adds the checklist proposed in #1458. It's roughly based on the checklist in the pull request template of https://github.com/htdvisser/template.

Closes #1458

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -28,3 +28,13 @@ implementation proposal in the referenced issues.
 -->
 
 ...
+
+#### Checklist
+<!-- Make sure that this pull request is complete. -->
+
+- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
+- [ ] Compatibility: The changes do not break existing API, database or configuration.
+- [ ] Testing: The changes are covered with unit tests. The changes have been tested by hand.
```
Comment (rvolosatovs) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -28,3 +28,13 @@ implementation proposal in the referenced issues.
 -->
 
 ...
+
+#### Checklist
+<!-- Make sure that this pull request is complete. -->
+
+- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
+- [ ] Compatibility: The changes do not break existing API, database or configuration.
```
Comment (rvolosatovs) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration.
```

**Comment by rvolosatovs** [comment]
@KrishnaIyer that is implied by `Testing` point.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (adriansmares) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think compatibility should also cover migrations:
`The changes are backwards compatible with existing API, database and configuration. In cases where the database schema has been changed, the migrations process should work correctly.`.

Comment (johanstokking) [reply .github/PULL_REQUEST_TEMPLATE.md]: This cannot and should not be exhaustive here as a checkbox. We have a compatibility commitment, just link to that, by checking that it aligns with the compatibility commitment.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -28,3 +28,13 @@ implementation proposal in the referenced issues.
...
+- [ ] Testing: The changes are covered with unit tests. The changes have been tested by hand.
+- [ ] Documentation: Relevant documentation is added or updated.
+- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to the changelog.
```
Comment (johanstokking) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add link

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -28,3 +28,13 @@ implementation proposal in the referenced issues.
...
+- [ ] Commits: Commit messages follow pattern, no fixup commits left.
```
Comment (johanstokking) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add link

**Comment by htdvisser** [comment]
@johanstokking: I'm afraid that adding links makes the template a bit too messy, so I just referred to the filenames.

**Pull request merged**